ipatests/test_integration/test_automember.py

```diff
@@ -0,0 +1,353 @@
+#
+# Copyright (C) 2019  FreeIPA Contributors see COPYING for license
+#
+"""This covers tests for automember feature
+"""
+from __future__ import absolute_import
+
+import textwrap
+
+from tempfile import NamedTemporaryFile
+from ipatests.test_integration.base import IntegrationTest
+from ipatests.pytest_ipa.integration import tasks
+
+
+class TestAutomember(IntegrationTest):
+    """
+    Tests of Automember feature.
+    Covers unmembering feature described in
+    https://pagure.io/389-ds-base/issue/50077
+    """
+    topology = 'line'
+
+    def user_automember_setup(self, group, user, first, last):
+        """
+        addition of user entry
+        """
+        cmdlist = [['ipa', 'group-add', '--desc=%s' % group, group],
+                   ['ipa', 'automember-add', '--type=group', group],
+                   ['ipa', 'automember-add-condition',
+                    '--key=destinationindicator', '--type=group',
+                    '--inclusive-regex=%s' % group, group],
```

Comment:
The 3 first commands can be put in a separate function (they are repeated for the 2nd group):
```def automember_group_add(self, group)```
that you can call during test initialization:
```
self.automember_group_add(group1)
self.automember_group_add(group2)
self.user_automember_setup(user1,group1)
```
Same thing for group deletion, you can write a function
```def automember_group_del(self, group)```
that will call automember-del and group-del.